pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -541,7 +541,19 @@ func (c *Controller) GetIstioServiceAccounts(hostname model.Hostname, ports []in
 // AppendServiceHandler implements a service catalog operation
 func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
 	c.services.handler.Append(func(obj interface{}, event model.Event) error {
-		svc := *obj.(*v1.Service)
+		svc, ok := obj.(*v1.Service)
+		if !ok {
+			tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				log.Errorf("Couldn't get object from tombstone %#v", obj)
+				return nil
```

Comment:
Do we need to log this? Too much noise ?

Reply:
@rshriram Not too many such logs. And kubernetes also log such message. This only can be produced when a delete event lost.  If you feel strongly, I can delete this log.